pkg/controller/command/vdri/command.go

```diff
@@ -101,6 +136,132 @@ func (o *Command) CreatePublicDID(rw io.Writer, req io.Reader) command.Error {
 	return nil
 }
 
+// SaveDID saves the did doc to the store
+func (o *Command) SaveDID(rw io.Writer, req io.Reader) command.Error {
+	request := &DIDExt{}
+
+	err := json.NewDecoder(req).Decode(&request)
+	if err != nil {
+		logutil.LogInfo(logger, commandName, saveDIDCommandMethod, "request decode : "+err.Error())
+
+		return command.NewValidationError(InvalidRequestErrorCode, fmt.Errorf("request decode : %w", err))
+	}
+
+	if request.Name == "" {
+		logutil.LogDebug(logger, commandName, saveDIDCommandMethod, errEmptyDIDName)
+		return command.NewValidationError(SaveDIDErrorCode, fmt.Errorf(errEmptyDIDName))
+	}
+
+	didDoc := &did.Doc{}
+	err = json.Unmarshal([]byte(request.DidDocument), didDoc)
+
+	if err != nil {
+		logutil.LogError(logger, commandName, saveDIDCommandMethod, "unmarshal did doc: "+err.Error())
+		return command.NewValidationError(SaveDIDErrorCode, fmt.Errorf("unmarshal did doc: %w", err))
+	}
+
+	err = o.didStore.SaveDID(request.Name, didDoc)
+	if err != nil {
+		logutil.LogError(logger, commandName, saveDIDCommandMethod, "save did doc: "+err.Error())
+
+		return command.NewValidationError(SaveDIDErrorCode, fmt.Errorf("save did doc: %w", err))
+	}
+
+	command.WriteNillableResponse(rw, nil, logger)
+
+	logutil.LogDebug(logger, commandName, saveDIDCommandMethod, "success")
+
+	return nil
+}
+
+// GetDID retrieves the did from the store.
+func (o *Command) GetDID(rw io.Writer, req io.Reader) command.Error {
+	var request IDArg
+
+	err := json.NewDecoder(req).Decode(&request)
+	if err != nil {
+		logutil.LogInfo(logger, commandName, getDIDCommandMethod, err.Error())
+		return command.NewValidationError(InvalidRequestErrorCode, fmt.Errorf("request decode : %w", err))
+	}
+
+	if request.ID == "" {
+		logutil.LogDebug(logger, commandName, getDIDCommandMethod, errEmptyDIDID)
+		return command.NewValidationError(InvalidRequestErrorCode, fmt.Errorf(errEmptyDIDID))
+	}
+
+	didDoc, err := o.didStore.GetDID(request.ID)
+	if err != nil {
+		logutil.LogError(logger, commandName, getDIDCommandMethod, "get did doc: "+err.Error(),
+			logutil.CreateKeyValueString(didID, request.ID))
+
+		return command.NewValidationError(GetDIDErrorCode, fmt.Errorf("get did doc: %w", err))
+	}
+
+	docBytes, err := didDoc.JSONBytes()
```

Comment:
Do we need to get bytes? why can't we write it directly to response writer. You can use raw JSON as property type in response model.